pkg/registry/docker_credentials.go

```diff
@@ -96,6 +96,9 @@ func (r DockerConfigAuthenticator) Authenticate(ctx context.Context, cfg DockerC
 		case candidates <- AuthenticationToken{
 			Kind:  "Basic",
 			Token: auth,
+			Ref: AuthenticationSourceRef{
+				Provider: "docker",
```

Comment:
Should we distinguish `ImagePullSecretAuthenticator` and `DockerConfigFileAuthenticator`?

```suggestion
				Provider: r.Provider,
```

like

```golang

type DockerConfigAuthenticator {
    Provider string
[...]
}
```

and later in the initialization:

```golang
func NewAuthenticator(kubeletConfigFile, kubeletBinDir string, privateRegistryPaterns []string) Authenticators {

	fs := afero.NewOsFs()
	a := Authenticators{
		ImagePullSecretAuthenticator{
Provider: "ImagePullSecrets",
},
	}
	if kubeletConfigFile != "" && kubeletBinDir != "" {
		a = append(a, KubeletAuthenticator{fs: fs, scheme: newScheme(), BinDir: kubeletBinDir, Config: kubeletConfigFile})
	} else {
		log.DefaultLogger.Info("no kubelet config file or bin dir provided, won't use kubelet authentication")
	}
	a = append(a,
		ContainerDAuthenticator{fs: fs},
		DockerConfigFileAuthenticator{

		ImagePullSecretAuthenticator{
Provider: "ImagePullSecrets",
},
fs: fs,
},
		AnonymousAuthenticator{
			PrivateRegistryPatterns: cleanRegistryPatterns(privateRegistryPaterns),
		},
	)
	return a
}


```

Reply:
sure :)

Reply:
i added the changes, wdyt?